src/services/datalake-storage.ts

```diff
@@ -0,0 +1,109 @@
+import { basename } from 'node:path';
+import { Readable } from 'node:stream';
+
+import {
+  DataLakeDirectoryClient,
+  DataLakeFileClient,
+  DataLakeFileSystemClient,
+  DataLakeServiceClient,
+  DirectoryCreateIfNotExistsResponse,
+  FileUploadResponse,
+  PathDeleteIfExistsResponse,
+  StorageSharedKeyCredential
+} from '@azure/storage-file-datalake';
+
+import { FileStore } from '../config/file-store.enum';
+import { StorageService } from '../interfaces/storage-service';
+import { logger as parentLogger } from '../utils/logger';
+
+const logger = parentLogger.child({ module: 'DataLake' });
+
+interface DataLakeConfig {
+  url: string;
+  accountName: string;
+  accountKey: string;
+  fileSystemName: string;
+}
+
+export default class DataLakeStorage implements StorageService {
+  private readonly serviceClient: DataLakeServiceClient;
+  private readonly fileSystemClient: DataLakeFileSystemClient;
+
+  public constructor(config: DataLakeConfig) {
+    const { url, accountName, accountKey, fileSystemName } = config;
+
+    const sharedKeyCredential = new StorageSharedKeyCredential(accountName, accountKey);
+    this.serviceClient = new DataLakeServiceClient(url, sharedKeyCredential);
+    this.fileSystemClient = this.serviceClient.getFileSystemClient(fileSystemName);
+  }
+
+  public getType(): FileStore {
+    return FileStore.DataLake;
+  }
+
+  public getServiceClient(): DataLakeServiceClient {
+    return this.serviceClient;
+  }
+
+  private getDirectoryClient(directory: string): DataLakeDirectoryClient {
+    return this.fileSystemClient.getDirectoryClient(directory);
+  }
+
+  private getFileClient(filename: string, directory: string): DataLakeFileClient {
+    return this.getDirectoryClient(directory).getFileClient(filename);
+  }
+
+  public async createDirectoryIfNotExists(directory: string): Promise<DirectoryCreateIfNotExistsResponse> {
+    return this.getDirectoryClient(directory).createIfNotExists();
+  }
+
+  public async saveBuffer(filename: string, directory: string, content: Buffer): Promise<FileUploadResponse> {
+    logger.debug(`Uploading file '${filename}' to datalake as buffer`);
+    await this.createDirectoryIfNotExists(directory);
```

Comment:
Rather than having to manually create the directory, just create it on the fly if it doesn't exist when saving.

This might mean a few more requests to datalake service but it means we don't have to think about it outside of this class.